src/main/java/edu/bu/metcs673/trackr/controller/TransactionController.java

```diff
@@ -0,0 +1,90 @@
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@Validated
+@RestController
+@RequestMapping("/api/v1/transaction")
+public class TransactionController {
+    @Autowired
+    private TrackrUserService trackrUserService;
+
+    @Autowired
+    private BankAccountService bankAccountService;
+
+    @Autowired
+    private TransactionService transactionService;
+
+
+    @PostMapping
+    public ResponseEntity<Transaction> createTransaction(@Valid @RequestBody TransactionDTO transactionInput) {
+
+        // pull username from JWT token, find corresponding user record
+        String username = SecurityContextHolder.getContext().getAuthentication().getPrincipal().toString();
+        TrackrUser user = trackrUserService.findByUsername(username);
+
+        BankAccount bankAccount = bankAccountService.findBankAccountByIdAndUserId(transactionInput.getBankAccountId(), user.getId());
+
+        // create a new bank account record associated to the user making the API request.
+        Transaction transaction = transactionService.createTransaction(transactionInput, bankAccount);
+
+        return new ResponseEntity<Transaction>(transaction, HttpStatus.OK);
+    }
+
+    @GetMapping("/find")
+    public ResponseEntity<Transaction> findTransactionById(@RequestParam("transactionId") long transactionId, @RequestParam("bankAccountId") long bankAccountId) {
+        // pull username from JWT token
+        String username = SecurityContextHolder.getContext().getAuthentication().getPrincipal().toString();
+        TrackrUser user = trackrUserService.findByUsername(username);
+
+        BankAccount bankAccount = bankAccountService.findBankAccountByIdAndUserId(bankAccountId, user.getId());
+
+        Transaction transaction = transactionService.findTraByIdAndBankAccountId(transactionId, bankAccount.getId());
+        return new ResponseEntity<Transaction>(transaction, HttpStatus.OK);
+
+    }
+
+    @GetMapping("/findAll")
+    public ResponseEntity<List<Transaction>> findTransactionById(@RequestParam("bankAccountId") long bankAccountId) {
+        // pull username from JWT token
+        String username = SecurityContextHolder.getContext().getAuthentication().getPrincipal().toString();
+        TrackrUser user = trackrUserService.findByUsername(username);
+
+        BankAccount bankAccount = bankAccountService.findBankAccountByIdAndUserId(bankAccountId, user.getId());
+
+        List<Transaction> transactions = transactionService.findAllTraByBankAccountId(bankAccount.getId());
+
+        return new ResponseEntity<List<Transaction>>(transactions, HttpStatus.OK);
+    }
+
+    @PostMapping("/modify")
+    public ResponseEntity<Transaction> modifyTransaction(@RequestParam("transactionId") long transactionId, @RequestBody TransactionDTO transactionInput) {
```

Comment:
I would suggest changing `transactionId` to be a path variable instead.  Also updating the mapping so it would be `@PostMapping("/modify/{transactionId}")`. There is an example in the BankAccountController.

Reply:
Please update the return type of this method to be `ResponseEntity<GenericApiResponse>` and provide some message and the transaction object. There is an example in the `BankAccountController`

Reply:
Ok, have done

Reply:
Sorry, I just find that 
you also use  `return new ResponseEntity<BankAccount>(bankAccount, HttpStatus.OK)` ```
in public ResponseEntity<BankAccount> modifyBankAccount(@Valid @RequestBody BankAccountDTO bankAccountInput,
                                                         @PathVariable(value = "id") long id)
``` method

Reply:
OK, have done